meson.build

```diff
@@ -1,4 +1,4 @@
-project('rizin', 'c',
+project('rizin', ['c', 'cpp'],
```

Comment:
I do not like this since it adds a strict dependency on having a C++ compiler for the entire project. But without it, meson will not build anything in C++. It should depend on the `enable_cpp_test` option but we can't check that before the `project()` call. @ret2libc maybe you have an idea here?

Reply:
If we create just a separate meson.build for the cpp tests?

Reply:
Agree, we could even put it in the `.github/workflows` since it's just for the CI.

Reply:
It's also possible to run `meson rewrite kwargs set project / language "cpp"` or something like that.